src/main/java/org/spongepowered/api/text/translation/dictionary/bundle/MultiSourceResourceBundleTranslationDictionary.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.translation.dictionary.bundle;
+
+import com.google.common.collect.Maps;
+import org.spongepowered.api.text.translation.dictionary.CachingTranslationDictionary;
+import org.spongepowered.api.text.translation.dictionary.TranslationDictionarySourceResolver;
+import org.spongepowered.api.text.translation.locale.Locales;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.MissingResourceException;
+import java.util.Optional;
+import java.util.ResourceBundle;
+import java.util.function.Function;
+
+/**
+ * A {@link ResourceBundleTranslationDictionary} with multiple resource bundle sources.
+ *
+ * @see SimpleResourceBundleTranslationDictionary
+ */
+public class MultiSourceResourceBundleTranslationDictionary extends SimpleResourceBundleTranslationDictionary
```

Comment:
This is a multi-source translation dictionary. There can be a primary classpath `ResourceBundle`, and plugins can add (to the resolver) extensions such as `PropertyResourceBundle`. The `Locale` is required to resolve file paths: `my/path/translations/{LOCALE}.properties` for example.